src/api/v1/donate.go

```diff
@@ -232,7 +234,23 @@ func handlePaymentSucceeded(c echo.Context, event *stripe.WebhookEvent, payment
 	return c.NoContent(http.StatusOK)
 }
 
-func handleChargeSucceeded(c echo.Context, event *stripe.WebhookEvent, charge *upstreamstripe.Charge) error {
+func handleChargeUpdated(c echo.Context, event *stripe.WebhookEvent, charge *upstreamstripe.Charge) error {
+	// Filter for an event where the charge has succeeded and the balance_transaction is present
+	if charge.BalanceTransaction == nil {
+		// The new stripe fast_automatic_capture_beta doesn't necessarily include the balance_transaction until later,
+		// check if this is the event that added it to the charge or if we are handling too early.
+		return c.String(http.StatusUnprocessableEntity, "Cannot handle a charge without a balance_transaction")
+	}
+	if charge.Status != "succeeded" {
+		return c.String(http.StatusUnprocessableEntity, "Cannot handle a charge that has not succeeded")
+	}
+	if charge.Refunded {
+		return c.String(http.StatusUnprocessableEntity, "Cannot handle a charge that has been refunded")
+	}
+	if charge.TransferGroup != "" {
+		return c.String(http.StatusUnprocessableEntity, "Cannot handle a charge that has already been distributed")
```

Comment:
Is this reliable? This check assumes that calling `transfer.New()` with `SourceTransaction` set to the charge's ID will update the charge's transfer group.

This is intended as a way to check if our `DistributeDonation` func has already been run against this charge, since we don't want to run it twice for a given charge.